src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementProductTable.tsx

```diff
@@ -68,8 +68,8 @@ const ReimbursementProductTable: React.FC<ReimbursementProductTableProps> = ({
     }
   });
 
-  const onCostBlurHandler = (value: number, name: string) => {
-    setValue(name, value.toFixed(2));
+  const onCostBlurHandler = (value: number, index: number) => {
+    setValue(`reimbursementProducts.${index}.cost`, parseInt(value.toFixed(2)));
```

Comment:
parseInt should be parseFloat here so we don't lose the two decimals points of info